specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,188 @@
+  - In the [csproj](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-3635dce04cb0a44d829f97259dce31b8f2abf8d0e80a4c5ccf9506babc93a78aR38), the Build Action for this is Content (not EmbeddedResource), `copyToOutput=true`, `IncludeInPackage=true`. This results in the resx file being included in the NuGet, in [contentFiles/](https://learn.microsoft.com/en-us/nuget/reference/nuspec#using-the-contentfiles-element-for-content-files)
+- [`Components\NimbleThemeProvider.razor`](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-935610f630ad061236774e447badf29f08dd1467f6c4c6e45ead7214641afe6d): Add code/markup for labels: i.e. `label-number-field-increment="@LabelNumberFieldIncrement"` in the Razor file, `[Parameter] public string? LabelNumberFieldIncrement { get; set; }` in the C# code. We may need to fully codegen the Razor file to add the attribute bindings.
+
+Once clients add a reference to the NuGet, `NimbleResources\NimbleLabels.resx` will appear in their project as a link (to that file in the NuGet). At that point, the remaining process is somewhat convoluted, and would need documentation. The prototype branch has done some of this in `Demo.Shared` and `Demo.Server`:
+- Browse to that resx file on disk, and manually copy it into the app project directory, in `Resources\`
+- Add it to the Blazor project (as Build Action = Embedded Resource). Then, open it in the resx editor, and set Access Modifier = Public or Internal (to generate the `.designer.cs` file)
+- To translate/localize the resources, clone the resx file (e.g. `NimbleLabels.fr.resx`) and translate the values in it. The app also needs to follow the [Microsoft Blazor localization docs](https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-7.0&pivots=server) in terms of opting into localization, configuring the supported locales, etc.
+- To consume the resources, in the Razor file using the NimbleThemeProvider, add:
+```cs
+[Inject]
+internal IStringLocalizer<NimbleLabels>? LabelProvider { get; set; }
+```
+Then bind to the theme provider labels in the Razor template:
+```xml
+<NimbleThemeProvider Theme="Theme" LabelNumberFieldDecrement="@LabelProvider!["labelNumberFieldDecrement"]" LabelNumberFieldIncrement="@LabelProvider!["labelNumberFieldIncrement"]">
+```
+
+The process above isn't great. When updated versions of Nimble Blazor are released, if they have new labels, the consuming app would need to repeat the process of manually copying the resx, and translating it. They'd also need to manually add the new label properties in their Razor file using the NimbleThemeProvider.
+
+If we have any clients that will be using Nimble Blazor and non-English locales we should probably do additional research to see if we can come up with a more seamless approach.
```

Comment:
I agree this seems pretty clunky. One idea for now would be to update this doc to say "we don't plan to do anything for Blazor yet" but list this research as an alternative. Another idea would be to post on the Web UI Blazor Teams channel and tag a few devs to ask whether their apps are localized and see if it's ok for us to defer this.

Reply:
Added the proposal that codegens a .resx file to Alternatives since it's still not really seamless.

Current proposal is to just have the Razor components corresponding to the `nimble-i18n-*` providers. Blazor clients would need to manually localize and set the properties on those. Updated the HLD based on that approach (and the updated prototype also just has that).